From 48596709d8ab59727b79a5c6db33ebb251c36543 Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Date: Thu, 19 Nov 2015 17:44:25 +0100
Subject: [PATCH] Avoid in6_addr redefinition
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Due to both <netinet/in.h> and <linux/in6.h> being included, the
in6_addr is being redefined: once from the C library headers and once
from the kernel headers. This causes some build failures with for
example the musl C library:

In file included from ../include/linux/xfrm.h:4:0,
                 from xfrm.h:29,
                 from ipxfrm.c:39:
../include/linux/in6.h:32:8: error: redefinition of ‘struct in6_addr’
 struct in6_addr {
        ^
In file included from .../output/host/usr/x86_64-buildroot-linux-musl/sysroot/usr/include/netdb.h:9:0,
                 from ipxfrm.c:34:
.../output/host/usr/x86_64-buildroot-linux-musl/sysroot/usr/include/netinet/in.h:24:8: note: originally defined here
 struct in6_addr
        ^

In order to fix this, use just the C library header <netinet/in.h>.

Original patch taken from
http://git.alpinelinux.org/cgit/aports/tree/main/iproute2/musl-fixes.patch.

Note: Modified for support backport to linux_libc_headers_3.18.131, to
match the linux_quic_3.18.131. This is for building kernel 3.18.131
with Yocto 2.5, which has a preferred libc header of 4.15.x

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
[Gustavo: drop ipt_kernel_headers.h chunk since no longer necessary]
Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>
Modified-by: Steven MacDonald <smacdonald@sierrawireless.com>
---
 include/uapi/linux/if_bridge.h       | 1 -
 include/linux/netfilter.h            | 2 --
 2 files changed, 3 deletions(-)

diff --git a/include/uapi/linux/if_bridge.h b/include/uapi/linux/if_bridge.h
index ee197a3..f823aa4 100644
--- a/include/uapi/linux/if_bridge.h
+++ b/include/uapi/linux/if_bridge.h
@@ -15,7 +15,6 @@

 #include <linux/types.h>
 #include <linux/if_ether.h>
-#include <linux/in6.h>

 #define SYSFS_BRIDGE_ATTR	"bridge"
 #define SYSFS_BRIDGE_FDB	"brforward"
diff --git a/include/linux/netfilter.h b/include/linux/netfilter.h
index b71b4c9..3e4e6ae 100644
--- a/include/linux/netfilter.h
+++ b/include/linux/netfilter.h
@@ -4,8 +4,6 @@
 #include <linux/skbuff.h>
 #include <linux/net.h>
 #include <linux/if.h>
-#include <linux/in.h>
-#include <linux/in6.h>
 #include <linux/wait.h>
 #include <linux/list.h>
 #include <linux/static_key.h>
-- 
2.6.3


